### DISCUSSION TITLE
Push notification not working when app is in foreground

### DISCUSSION THREAD

**Discussion by umairrafiq1133** [ionic-team/capacitor#3623] [Q&A]
I have recently migrated to capacitor from cordova. I am having issues with push notification when app is in foreground.
In cordova-plugin-firebasex you have need to add notification_foreground as true in additional data and you will get a push

I am not finding any way to force show push notification when app is in foreground. 

https://capacitorjs.com/docs/apis/push-notifications#push-notifications-appearance-in-foreground

I have followed the instructions and added this code in `capacitor.config.json`

```
"plugins": {
  "PushNotifications": {
    "presentationOptions": ["badge", "sound", "alert"]
  }
}
```

But I dont get push notification alert when app is in foreground.

In background everything works as expected. Need help..!!!

**Comment by umairrafiq1133** [comment]
@dwieeb Can you help? Or mention someone who can?

**Reply by olehtymoshenko** [reply]
Did you solve the problem? I have the same

**Comment by alex-migwi** [comment]
You can check the answer here and see if that  helps you out #3286

**Reply by kiranrayala77** [reply]
> You can check the answer here and see if that helps you out #3286

pushNotificationReceived is. not  able to 
trigged when the. app is in foreground

**Comment by kiranrayala77** [comment]
Im also facing same. issue

**Comment by kiranrayala77** [comment]
> I have recently migrated to capacitor from cordova. I am having issues with push notification when app is in foreground. In cordova-plugin-firebasex you have need to add notification_foreground as true in additional data and you will get a push
> 
> I am not finding any way to force show push notification when app is in foreground.
> 
> https://capacitorjs.com/docs/apis/push-notifications#push-notifications-appearance-in-foreground
> 
> I have followed the instructions and added this code in `capacitor.config.json`
> 
> ```
> "plugins": {
>   "PushNotifications": {
>     "presentationOptions": ["badge", "sound", "alert"]
>   }
> }
> ```
> 
> But I dont get push notification alert when app is in foreground.
> 
> In background everything works as expected. Need help..!!!

Im also facing same issue can you please help me on this
